src/Domain/Repositories/ApprenticeshipQueryRepository.cs

```diff
@@ -139,6 +138,25 @@ public async Task<IEnumerable<DataTransferObjects.Apprenticeship>> GetAll(long u
          return pendingStartDateChange;
      }
 
+     public async Task<bool?> GetPaymentStatus(Guid apprenticeshipKey)
+     {
+	     bool? paymentsFrozen = null;
+
+	     try
+	     {
+            paymentsFrozen = await DbContext.Apprenticeships
+	            .Where(x => x.Key == apprenticeshipKey)
+	            .Select(x => x.PaymentsFrozen)
+	            .SingleOrDefaultAsync();
```

Comment:
Should this use Single instead of SingleOrDefault, if the record is not found it will successfully return a nullable result without throwing.

Reply:
good spot, will change